### ISSUE TITLE
Doesn't restore maximized window state

### ISSUE THREAD

**Issue by sergsako** [sumatrapdfreader/sumatrapdf#4169]
**SumatraPDF version**
pre-release 3.6.16004
but stable 3.5.2 is not affected

**Describe the bug**
When I open a PDF file restored from the Recycle Bin, SumatraPDF forgets its window size and opens half-screen wide

**To Reproduce**
1. Open a PDF from a folder by double-click.
2. SumatraPDF respects its last used window size (maximized in my case).
3. Close the document tab, close the program window, delete the PDF file to the Recycle Bin (using the Windows Explorer, but the same happens if the new command of the pre-release to delete a document is used).
4. Restore the PDF file from the Recycle Bin to the initial folder and open the PDF by double-click.
5. SumatraPDF starts with the PDF doc, but in a half-width window.

**Expected behavior**
SumatraPDF should remember its window size.

**File that reproduces the problem**
Not related to a specific document.

**Screenshots**
No screenshots.

**Additional context**
Windows 10 x64 21H2. SumatraPDF setting: option: default zoom: fit width.

**Comment by kjk** [comment]
Can't reproduce. Is it with tabs enabled?

Can you make a video?

**Comment by sergsako** [comment]
Yes, tabs are enabled. Portable SumatraPDF pre-release was used, only the new EXE was put to a usual folder of the program, the sumatrapdfcache folder and SumatraPDF-settings.txt file of the previous version were removed, then after the first start the window was maximized and the "default zoom: fit width" setting was made, no more. 
No, I am not able to make a video. Sorry, if the problem is irreproducible.

**Comment by sergsako** [comment]
https://github.com/sumatrapdfreader/sumatrapdf/assets/106967914/379c4adc-c89b-44d3-8a16-6de9adfe6133

I made a video, somehow.

**Comment by kjk** [comment]
Thanks, it's not related to deleting a file, restoring maximized state doesn't work.

Simpler repro:
* open a file
* maximize window
* close the file
* close the window
* double-click the file

It should open maximized but instead of flashes as if were maximized and then it seems it gets resized. At the same time the maxmize button shows as if the window was maximized.

So I suspect we first maxmize the window and then resize / reposition the window.

**Comment by kjk** [comment]
I can fix it by adding:
```
        args->noPlaceWindow = true;
```
in SumatraPDF.cpp@1856 but then it causes to fullscreen even if window wasn't fullscreen when closed.

**Comment by reccate** [comment]
https://github.com/sumatrapdfreader/sumatrapdf/issues/4036

**Comment by kjk** [comment]
Yeah, that logic is complicated and I frankly don't truly understand it anymore.

**Comment by reccate** [comment]
I can't figure out what the function of the code at line 1257 to 1275 is for.

Remove this block of code work for me.

```c++
    bool shouldPlace = args->isNewWindow || args->placeWindow && fs;
    if (args->noPlaceWindow) {
        shouldPlace = false;
    }
    if (shouldPlace) {
        if (args->isNewWindow && fs && !fs->windowPos.IsEmpty()) {
            // Make sure it doesn't have a position like outside of the screen etc.
            Rect rect = ShiftRectToWorkArea(fs->windowPos);
            // This shouldn't happen until !win.IsAboutWindow(), so that we don't
            // accidentally update gGlobalState with this window's dimensions
            MoveWindow(win->hwndFrame, rect);
        }
        if (args->showWin) {
            ShowWindow(win->hwndFrame, showType);
        }
        if (win) {
            UpdateWindow(win->hwndFrame);
        }
    }
```

**Comment by kjk** [comment]
Well, the reason I couldn't fix this bug so far is because the logic for opening files became very complex over time (unfortunately).

The purpose of this code is what it says: if a window happens to outside of screen (e.g. you had window on the right side, you close  sumatra, you lower resolution of the screen, you re-open sumatra, position of windows is restored and now invisible) it repositions the window so it's fully visible.

I'm sure removing that fixes the one case you're testing but the reason this code is complex is because there are many ways to open a file (restore session at startup, dde commands, force reload with 'r', file / open, drag&drop, lazy loading) and just removing that part likely breaks some other scenario.

**Comment by reccate** [comment]
Thanks for your explanation.

Your Solution work flawless for me.
```c++
 args->noPlaceWindow = true;
```

**Comment by kjk** [comment]
should be fixed by https://github.com/sumatrapdfreader/sumatrapdf/commit/159ad122e8ea13bfbb590be1c2c9c1e856eea1ba
